client/state/selectors/get-site-default-post-format.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getRawSite, getSiteOption } from 'state/sites/selectors';
+import { getSiteSettings } from 'state/site-settings/selectors';
+
+/**
+ * Returns the default post format of a site.
+ * Returns null if the site is unknown.
+ *
+ * @param  {Object}    state   Global state tree
+ * @param  {Number}    siteId  The ID of the site we're querying
+ * @return {?String}           The default post format of that site
+ */
+export default function getSiteDefaultPostFormat( state, siteId ) {
+	const site = getRawSite( state, siteId );
+	if ( ! site ) {
+		return null;
+	}
+
+	let defaultPostFormat = get( getSiteSettings( state, siteId ), 'default_post_format' );
+	if ( ! defaultPostFormat ) {
+		defaultPostFormat = getSiteOption( state, siteId, 'default_post_format' );
+	}
+
+	if ( ! defaultPostFormat || defaultPostFormat === '0' ) {
```

Comment:
>Yes, but since we're now migrating away from `sites-list`, the `settings` will not be available in the `site` object.

Right, but there's a hypothetical situation where we have the site's settings received from `GET /sites/%s/settings` but _not_ the `GET /sites/%s` data, in which case I'd expect the data from the prior to be returned. But with the implementation here, it'll never reach that point because the site hasn't yet been received. Or at least based on the intent of the selector, I'd expect this to be the flow of logic (even if in practice this would never occur).

>But if we have information about the site, it would make sense to return the `standard` post

I agree with this. I think I was confused by my internal desire to remove the `if ( ! site ) {` condition at the top of the function, but with it, your reasoning makes sense. Generally, I just want to be sure that we don't return `standard` if the site data hasn't been received (since, for all we know, the default format is actually `aside`).